mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java

```diff
@@ -390,6 +406,18 @@ private Condition getMatchedFieldCondition(MatchField matchedField, String parti
     return condition(sql);
   }
 
+  private static String getComparisonValue(Filter.ComparisonPartType comparisonPartType) {
+    if (comparisonPartType == null) {
+      return "\"{partition}\".\"value\"";
+    } else {
+      if (comparisonPartType == ALPHANUMERICS_ONLY) {
+        return "regexp_replace(\"{partition}\".\"value\", '[^a-zA-Z0-9]', '', 'g')";
```

Comment:
```suggestion
        return "regexp_replace(\"{partition}\".\"value\", '[^[:alnum:]]', '', 'g')";
```
propose to apply the following to avoid replacement/lost of characters that are not included in the a-zA-Z range (e.g. cyryllic characters) in case field of existing record contains such